libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -40,16 +39,15 @@
 NSString* const kSFDefaultContentType = @"application/json";
 NSInteger const kSFRestErrorCode = 999;
 
-// singleton instance
-static SFRestAPI *_instance;
-static dispatch_once_t _sharedInstanceGuard;
 static BOOL kIsTestRun;
+static NSMutableDictionary *sfRestApiList = nil;
```

Comment:
Couldn't we replace this with SFSDKSafeMutableDictionary instead? Then all the guarded   blocks can be reduced?